serial/serialutil.py

```diff
@@ -559,7 +559,7 @@ def readinto(self, b):
     # context manager
 
     def __enter__(self):
-        if not self.is_open:
+        if self.port and not self.is_open:
```

Comment:
I think the check should be against `None`, like this
```suggestion
        if self._port is not None and not self.is_open:
```

Reply:
Yeap. self._port is better than self.port